clang/lib/Sema/SemaTemplateInstantiateDecl.cpp

```diff
@@ -2129,7 +2129,7 @@ Decl *TemplateDeclInstantiator::VisitFunctionDecl(
     Function = CXXDeductionGuideDecl::Create(
         SemaRef.Context, DC, D->getInnerLocStart(),
         InstantiatedExplicitSpecifier, NameInfo, T, TInfo,
-        D->getSourceRange().getEnd(), /*Ctor=*/nullptr,
+        D->getSourceRange().getEnd(), DGuide->getCorrespondingConstructor(),
```

Comment:
I'm a new developer, can someone with more experience in clang check if this can never cause a use after free.

Reply:
I don't believe this will, we're not deleting the deduction  guide.